### ISSUE TITLE
refactor autoteleport

### ISSUE THREAD

**Issue by daiagi** [kodadot/nft-gallery#9695]
## Current Situation

1. All of AutoTeleport state and function is bounded to `AutoTeleportActionButton`
2.  the chain of calls to composables starting from `useAutoTeleport` is messy and hard to read/ reason
3. for example fees is being drilled through 3 composables to affect `isReady` then is then being drilled up 3 composables, instead there should be a simple function where fees changes `autoTeleportStore.isReady` 

## What i would like to see
1. identify refs and computed refs that belong in store, move them to store
2. function over these state items can be in simple composables / util functions / store actions 
3. composables should have no args or only very simple args (`useIsChain` as an example for simple arg)
4. component can then use multiple composables or use store directly to get whatever it needs

**Comment by daiagi** [comment]
@hassnian  are you interested in taking this on?

**Comment by hassnian** [comment]
> @hassnian are you interested in taking this on?

sure , will do

**Comment by Temidayo32** [comment]
@daiagi I would like to work on this. Or is it only for the internal team?

**Comment by daiagi** [comment]
@Temidayo32 
Honestly i suggest you start with easier pieces to bite

this one is no small task

but, if you feel confident and ready for possibly many corrections to the PR, then please go ahead

**Comment by Temidayo32** [comment]
@daiagi Thanks for the response. I will give it a shot.

**Comment by daiagi** [comment]
type out `:wave:` to assign yourself

**Comment by Temidayo32** [comment]
👋

**Comment by exezbcz** [comment]
still p3 prio? @vikiival @hassnian

**Comment by vikiival** [comment]
Like from design perspective it works ok, probably code wise ot would need some love

**Comment by hassnian** [comment]
> Like from design perspective it works ok, probably code wise ot would need some love

yep, and refactoring this requires a lot more time than it seems and we may introduce new bugs in the process

I'd leave it at `p3` for now, as our resources would be better spent addressing other issues with higher prio.
